### PR TITLE
Logging and events

### PR DESCRIPTION
**Pull request by enxebre**
Improve logging and add events for the operator cluster status
Conventions:
- https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/logging.md
- if an error message is just printed, use uppercase
- if an error message is return by a function/method and passed on, use lowercase
- NOUN VERB STATE format when possible

### CODE REVIEW

File: cmd/machine-api-operator/helpers.go [hunk 1]
```diff
@@ -7,7 +7,7 @@ import (
 
 	"github.com/golang/glog"
 
-	"k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
```
Comment (ingvagabund) [inline cmd/machine-api-operator/helpers.go]: why?

File: cmd/machine-api-operator/start.go [hunk 2]
```diff
@@ -67,7 +67,7 @@ func runStartCmd(cmd *cobra.Command, args []string) {
 		Callbacks: leaderelection.LeaderCallbacks{
 			OnStartedLeading: run,
 			OnStoppedLeading: func() {
-				glog.Fatalf("leaderelection lost")
+				glog.Fatalf("leader election lost")
```
Comment (ingvagabund) [inline cmd/machine-api-operator/start.go]: uppercase `Leader`

File: cmd/machine-api-operator/helpers.go (hunk 1 above)
Comment (enxebre) [reply cmd/machine-api-operator/helpers.go]: goimports generates it

File: cmd/machine-api-operator/start.go (hunk 2 above)
Comment (frobware) [reply cmd/machine-api-operator/start.go]: Awkward one, but I tend to go with: https://github.com/golang/go/wiki/CodeReviewComments#error-strings

**Comment by frobware** [comment]
I think the "rules" would be simpler if it was lower _always.. With that simple rule there's no ambiguity. Code reviews go faster because there's only 1 rule. Error message are consistently formatted. It is also idiomatic Go [1].

[1] - https://github.com/golang/go/wiki/CodeReviewComments#error-strings

**Comment by enxebre** [comment]
/test integration
for debugging

**Comment by enxebre** [comment]
fixes integration https://github.com/openshift/machine-api-operator/pull/233

File: pkg/operator/status.go [hunk 3]
```diff
@@ -39,10 +41,18 @@ func (optr *Operator) statusProgressing() error {
...
+	if err != nil {
+		glog.Errorf("Failed to get or create Cluster Operator: %v", err)
+		return err
+	}
+
 	var message string
 	if !reflect.DeepEqual(desiredVersions, currentVersions) {
 		glog.V(2).Info("Syncing status: progressing")
 		message = fmt.Sprintf("Progressing towards %s", optr.printOperandVersions())
+		optr.eventRecorder.Eventf(co, v1.EventTypeNormal, "Status upgrade", message)
```
Comment (ingvagabund) [inline pkg/operator/status.go]: `Status upgraded` or `Status upgraging` maybe?

**Comment by ingvagabund** [comment]
Does it make sense to report `Status successfully updated` as well?

File: cmd/nodelink-controller/main.go [hunk 4]
```diff
@@ -388,17 +388,17 @@ func (c *Controller) processNode(node *corev1.Node) error {
...
-		glog.V(3).Infof("copying label %s = %s", k, v)
+		glog.V(3).Infof("Copying label %s = %s", k, v)
 		modNode.Labels[k] = v
 	}
 
 	addTaintsToNode(modNode, matchingMachine)
 
 	if !reflect.DeepEqual(node, modNode) {
-		glog.V(3).Infof("node has changed, updating")
+		glog.V(3).Infof("Node has changed, updating")
```
Comment (frobware) [inline cmd/nodelink-controller/main.go]: If we're going to log we might as well indicate _which_ node.

**Pull request merged**